app/src/main/java/unipd/se18/ocrcamera/MainApplication.java

```diff
@@ -0,0 +1,96 @@
+package com.example.mattia.fotocamera;
+
+import android.app.Application;
+import android.content.res.AssetManager;
+import android.util.Log;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public class MainApplication extends Application {
+
+    public static MainApplication instance = null;
+
+    public void onCreate(){
+        super.onCreate();
+        //start copy file here
+        instance=this;
+        copyTessDataForTextRecognizor();
+    }
+
+    /**
+   	* @author Fasan Giovanni
+    * @param none
+   	* @return String of tess's data path
+    */
+
+    private String tessDataPath(){
+        return MainApplication.instance.getExternalFilesDir(null)+"/tessdata/";
+    }
+
+    /**
+    * @author Fasan Giovanni
+    * @param none
+    * @return String of tess's data parent directory
+    */
+
+    public String getTessDataParentDirectory(){
+        return MainApplication.instance.getExternalFilesDir(null).getAbsolutePath();
+    }
+
+    /**
+    * @author Fasan Giovanni
+    * @param none
+    * @return void
+    * open and read tessdata file for recognizer the text
+    */
+
+    private void copyTessDataForTextRecognizor(){
+        Runnable run = new Runnable() {
+            @Override
+            public void run() {
+                AssetManager assetManager = MainApplication.instance.getAssets();
+                OutputStream out = null;
+                try{
+                    Log.d("MainApplication", "CopyTessDataForTextRecognizor");
+                    InputStream in = assetManager.open("ita.traineddata");
```

Comment:
Is "ita.traineddata" a String you chose? To avoid hardcoding could be better to define a final variable at the beginning of the method.

Reply:
Yes you're right it's actually hardcoding.
ita.traineddata is the name of the library file.